docs/imageMgmtAPI.rst

```diff
@@ -0,0 +1,337 @@
+by layering the required images with their desired versions at run-time.
+
+Key Requirements
+----------------
+1. The image created based on azkaban binaries, platform binaries etc. as part of the release process must be available
+   for Azkaban to consume and use.
+#. Image owners/developers should be able to roll out their images independently without having dependency on
+   Azkaban codebase.
+#. Image roll outs to be independent of Azkaban release.
+#. Maintain a single source of truth for images and their associated metadata.
+#. REST API for image rollouts and rampups to allow for a self-service model.
+
+REST API EndPoints
+------------------
+Azkaban web server will host these APIs for each Azkaban cluster. Initially to start with there will be image management CRUD APIs and percentage wise image ramp up API.
+API for populating new image type
+This API will be used to populate newly created image types. As a result of invocation of this API, a new image type will be inserted in the image_types table. One owner and role (admin by default) can be specified during image creation. Separate ownerships API will be provided later to manage image ownerships. Ownership information will be stored in the image_ownerships table.
+
+/imageTypes
+***********
+
+.. _create-image-type:
+
+Register new image type (Create)
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+- **Method:** POST
+- **Request URL:** /imageTypes
+- **Request Body:**
+
++-----------------+-------------+--------------------------------------+
+|   Field Name    |     Type    |            Description               |
++=================+=============+======================================+
+|  imageType      |   String    |  Image Type Name (PK)                |
++-----------------+-------------+--------------------------------------+
+|  description    |   String    |  Description of the image type       |
++-----------------+-------------+--------------------------------------+
+|  deployable     |   enum      |  IMAGE (docker image), JAR or TAR    |
++-----------------+-------------+--------------------------------------+
+|  ownerships     | JSON String | List of:                             |
+|                 |             | {"owner": <user-id>, "role": <role>} |
+|                 |             |                                      |
+|                 |             | Possible roles: "admin" or "member"  |
++-----------------+-------------+--------------------------------------+
+
+**Response:**
+
+.. code-block::
+
+  Status: 201 Created
+  Header -> Location: /imageTypes/{id}
+
+/imageVersions
+**************
+This endpoint is used to register or list image versions for a given image type. The API
+assumes that image type is already registered (Create API for /imageTypes was called before) with Azkaban.
+Also the image binary must already be uploaded to a location in artifactory, accessible to Azkaban.
```

Comment:
Should we clarify if this requirement is enforced ?
I.e Do we actually check if the image is available on artifactory when this API is invoked ?

Reply:
done.